Userland/DevTools/HackStudio/Dialogs/ProjectTemplatesModel.cpp

```diff
@@ -21,12 +23,23 @@ ProjectTemplatesModel::ProjectTemplatesModel()
     : m_templates()
     , m_mapping()
 {
-    auto watcher_or_error = Core::FileWatcher::watch(ProjectTemplate::templates_path());
+    auto watcher_or_error = Core::FileWatcher::create();
     if (!watcher_or_error.is_error()) {
         m_file_watcher = watcher_or_error.release_value();
         m_file_watcher->on_change = [&](auto) {
             update();
         };
+
+        auto watch_result = m_file_watcher->add_watch(
+            ProjectTemplate::templates_path(),
+            InodeWatcherEvent::Type::ChildCreated
+                | InodeWatcherEvent::Type::ChildDeleted);
```

Comment:
It feels a bit weird that we have to provide `InodeWatcherEvent::Type` here (a kernel API enum) instead of `Core::FileWatcher` having something "brand-labeled" for us.

Reply:
Sure, I do think it's a bit backwards too :^) That requires a little translation work on the `FileWatcher` side though.